### PR TITLE
Add Decomposition of Resource Scatter Add in TF Compiler

### PR DESCRIPTION
**Pull request by raikonenfnu**

### CODE REVIEW

**Comment by stellaraccident** [comment]
The code looks about right: is there a corresponding test update?

**Comment by raikonenfnu** [comment]
@stellaraccident not yet, I'll add it here https://github.com/tensorflow/tensorflow/blob/21c46e80edd69d352ae8ff01811e9631644f790f/tensorflow/compiler/mlir/tensorflow/tests/decompose_resource_ops.mlir :)

**Comment by raikonenfnu** [comment]
> The code looks about right: is there a corresponding test update?

Added :)

**Comment by raikonenfnu** [comment]
I am trying to fix my bug S.T AMD ROCm CI can pass, but I can't seem to access/view the logs on jenkins 🤔

**Comment by powderluv** [comment]
The link to the build failure is wrong. Poking around I was able to get your error log:

http://ml-ci.amd.com:21096/job/tensorflow/job/github-prs-upstream-master/job/AMD-ROCm-Community-CI-Build/view/change-requests/job/PR-50314/lastBuild/console

**Comment by raikonenfnu** [comment]
> The link to the build failure is wrong. Poking around I was able to get your error log:
> 
> http://ml-ci.amd.com:21096/job/tensorflow/job/github-prs-upstream-master/job/AMD-ROCm-Community-CI-Build/view/change-requests/job/PR-50314/lastBuild/console

Thanks @powderluv, after giving it an additional Builder for the TensorScatterAddOp seems to have done the trick locally at least :)

**Review by stellaraccident — APPROVED** [review]

File: tensorflow/compiler/mlir/tensorflow/ir/tf_generated_ops.td [hunk 1]
```diff
@@ -17964,6 +17964,11 @@ On GPU, if an out of bound index is found, the index is ignored.
 
   TF_DerivedOperandTypeAttr T = TF_DerivedOperandTypeAttr<0>;
   TF_DerivedOperandTypeAttr Tindices = TF_DerivedOperandTypeAttr<1>;
+
+  let builders = [
```
Comment (stellaraccident) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_generated_ops.td:17968]: We need to check with some folks internally on whether local edits here make sense. May need to kick this back.

**Pull request merged**